From b40cb1d54c68ed91fedecef78bc6cf69bae62ef6 Mon Sep 17 00:00:00 2001
From: Lorenzo Pirritano <lpirritano@anaconda.com>
Date: Mon, 22 Jan 2024 21:02:41 -0600
Subject: [PATCH] disable xoshiro

https://github.com/david-cortes/isotree/blob/master/src/xoshiro.hpp 
is license CC 1.0 and Snowflake does not want this. 

We manually disable this as it seems to be optional, see 
https://github.com/david-cortes/isotree/blob/master/CMakeLists.txt#L33-L40
and
https://github.com/david-cortes/isotree/blob/master/setup.py#L333.

---
 setup.py | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/setup.py b/setup.py
index c96cb4f..9b14e00 100644
--- a/setup.py
+++ b/setup.py
@@ -330,8 +330,7 @@ setup(
                                 include_dirs=[np.get_include(), ".", "./src"],
                                 language="c++",
                                 install_requires = ["numpy", "pandas>=0.24.0", "cython", "scipy"],
-                                define_macros = [("_USE_XOSHIRO", None),
-                                                 ("NDEBUG", None),
+                                define_macros = [("NDEBUG", None),
                                                  ("_FOR_PYTHON", None),
                                                  ("CYTHON_EXTERN_C", 'extern "C"'),
                                                  ("NPY_NO_DEPRECATED_API", "NPY_1_7_API_VERSION")]
-- 
2.39.1

